code/modules/mob/living/status_procs.dm

```diff
@@ -448,91 +448,11 @@
 /mob/living/SetWeakened(amount, updating = 1, force = 0)
 	if((!!amount) == (!!weakened)) // We're not changing from + to 0 or vice versa
 		updating = FALSE
-	if(status_flags & CANWEAKEN || force)
+	if(((status_flags & CANWEAKEN) && !(has_trait(TRAIT_STUNIMMUNE))) || force)
 		weakened = max(amount, 0)
 		if(updating)
 			update_canmove()	//updates lying, canmove and icons
 
 /mob/living/AdjustWeakened(amount, bound_lower = 0, bound_upper = INFINITY, updating = 1, force = 0)
 	var/new_value = directional_bounded_sum(weakened, amount, bound_lower, bound_upper)
-	SetWeakened(new_value, updating, force)
-
-//
-//		DISABILITIES
-//
```

Comment:
Moved up above a level to mob/ in mob_traits.dm instead of mob/living for better compatibility with most procs & easier replacing of mutations